### PR TITLE
[`MBart`] Fix cross attention mask check

### LINKED ISSUE #21728: Error with shape-assertion regarding head_masks in mbart_modeling file
### System Info

file: https://github.com/huggingface/transformers/blob/v4.26.1/src/transformers/models/mbart/modeling_mbart.py
commit-hash: fd5cdaeea6eafac32e9d967327bfa3dc0e0d962d

### Who can help?

@ArthurZucker @younesbelkada

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

Call the forward method of MBartModel with:
- head_mask=None,
- decoder_head_mask=None,
 and 
- cross_attn_mask=torch.ones(mismatching_layer_size, number_of_heads)

### Expected behavior

**Expected Output:** The cross_attn_head_mask should be specified for X layers, but it is for Y.
**Actual Output:** 'NoneType' object has no attribute 'size'

**Potential fix:** The head_mask from line 1058 should probably reference the attn_mask from line 1053.

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

Fixes https://github.com/huggingface/transformers/issues/21728

The current `MBart` code leads to an error that is hard to interpret for users due to a possible typo as pointed out in https://github.com/huggingface/transformers/issues/21728 
To reproduce: 
```python
import torch
from transformers import MBartModel

input_ids = torch.LongTensor([[0, 1, 1, 0]])
model = MBartModel.from_pretrained("facebook/mbart-large-cc25")

head_mask = None

cross_attn_head_mask = torch.ones(1000, 1, 1, 1)
model(input_ids, head_mask=head_mask, cross_attn_head_mask=cross_attn_head_mask)
```

This PR fixes this

cc @sgugger

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thanks for the fix!
